### PR TITLE
Add Signet and testnet4 launch shortcuts for Windows

### PR DESCRIPTION
**Pull request by Sjors**
This makes it easier to launch Signet and testnet4 on Windows. Follows the same pattern as testnet.

Before:

testnet

After:
signet1
signet2

(the testnet4 icon is the same as testnet3, not in the above screenshot)

### CODE REVIEW

**Comment by MarcoFalke** [comment]
DrahtBot may have build results in two days. Though, keep in mind that they are untrusted.

**Comment by 1440000bytes** [comment]
Concept ACK

Tested it on Windows and it works as expected except the **color of logo** for signet. 

```
    Directory: C:\Users\Test\AppData\Roaming\Microsoft\Windows\Start Menu\Programs\Bitcoin Core


Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
-a----        18-10-2022     15:55            890 Bitcoin Core (64-bit).lnk
-a----        18-10-2022     15:55           1774 Bitcoin Core (signet, 64-bit).lnk
-a----        18-10-2022     15:55           1776 Bitcoin Core (testnet, 64-bit).lnk
-a----        18-10-2022     15:55            885 Uninstall Bitcoin Core (64-bit).lnk
```

image

**Comment by Sjors** [comment]
Looks like I need to set the correct yellow color.

**Review by jarolrod — COMMENTED** [review]
we have actual ico files under res for the main and testnet icons, you would need to introduce a bitcoin_signet.ico with the correct yellow hue applied under src/qt/res in order to introduce the correct icon.

Looking ahead, I'd like to move away from having these different launchers to being able to select the chain from within the gui.

**Comment by Sjors** [comment]
> Looking ahead, I'd like to move away from having these different launchers to being able to select the chain from within the gui

That might run into objections, see discussion in [#signet](https://github.com/bitcoin-core/gui/pull/414)

**Comment by luke-jr** [comment]
Most users won't know what "signet" means, and it isn't self-explanatory like "testnet"...

If you want a shortcut, please figure out a way to resolve this.

NACK as-is.

**Comment by fanquake** [comment]
What is the status of this? Seems very low priority, and not currently mergable.

**Comment by Sjors** [comment]
[running Guix build now, will mark as ready for review when that's done and I've tested it]

I added the word "test" to the application name.

Since [Bitcoin Inquisition](https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2022-September/020921.html) uses the default Signet to deploy several proposed soft forks, I think it's useful to offer easier access (which requires the Inquisition folks to rebase after this is merged).

See #8285 for additional context.

I generated the icon from the mainnet icon by applying the same hue and saturation adjustments:

https://github.com/bitcoin/bitcoin/blob/94070029fb6b783833973f9fe08a3a871994492f/src/qt/networkstyle.cpp#L22

Though the [-modulate](https://imagemagick.org/script/command-line-options.php#modulate) command uses percentage instead of degrees, so it was a bit confusing to convert:

```sh
convert  src/qt/res/icons/bitcoin.ico -modulate 100,87,119.4 src/qt/res/icons/bitcoin_signet.ico
```

**Comment by Sjors** [comment]
I think I found the right incantation, using the high res PNG version as input:

```
convert  src/qt/res/icons/bitcoin.png -modulate 100,87,119.4 -define icon:auto-resize="256,48,32,16" src/qt/res/icons/bitcoin_signet.ico
```

Will update screenshot and push the updated file after I finish my Guix build.

File: share/setup.nsi.in [hunk 1]
```diff
@@ -141,6 +142,7 @@ Section -un.post UNSEC0001
     Delete /REBOOTOK "$SMPROGRAMS\$StartMenuGroup\Uninstall $(^Name).lnk"
     Delete /REBOOTOK "$SMPROGRAMS\$StartMenuGroup\$(^Name).lnk"
     Delete /REBOOTOK "$SMPROGRAMS\$StartMenuGroup\@PACKAGE_NAME@ (testnet, 64-bit).lnk"
+    Delete /REBOOTOK "$SMPROGRAMS\$StartMenuGroup\@PACKAGE_NAME@ (signet, 64-bit).lnk"
```
Comment (Sjors) [inline share/setup.nsi.in:145]: Forgot to add `test ` here, pushing fix.

**Comment by 1440000bytes** [comment]
> Most users won't know what "signet" means, and it isn't self-explanatory like "testnet"...

Most users wont use it and it just helps devs and power users on windows. They know what signet means. Everyone else clicking on it is same as downloading a malware instead of bitcoin core which is not something this repo needs to solve. First match in search is always "bitcoin core (mainnet)" and even that would open other network if bitcoin.conf has other network set.

So this NACK makes no sense for windows users.


> I added the word "test" to the application name.
> 
> Since [Bitcoin Inquisition](https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2022-September/020921.html) uses the default Signet to deploy several proposed soft forks, I think it's useful to offer easier access (which requires the Inquisition folks to rebase after this is merged).
> 
> See #8285 for additional context.
> 
> I generated the icon from the mainnet icon by applying the same hue and saturation adjustments:
> 
> https://github.com/bitcoin/bitcoin/blob/94070029fb6b783833973f9fe08a3a871994492f/src/qt/networkstyle.cpp#L22
> 
> Though the [-modulate](https://imagemagick.org/script/command-line-options.php#modulate) command uses percentage instead of degrees, so it was a bit confusing to convert:
> 
> ```shell
> convert  src/qt/res/icons/bitcoin.ico -modulate 100,87,119.4 src/qt/res/icons/bitcoin_signet.ico
> ```
> 
> I added a screenshot in the PR description.
> 
> Screenshot_20230221_125320
> 
> The big icon looks correct, but the mini version in the search result does not. I probably need to change the ImageMagick command somehow. I could also try using the Testnet icon as the basis (and rotate the hue in the other direction).

NACK for "test signet"

`signet` already helps to know its NOT mainnet

**Comment by Sjors** [comment]
I don't see any harm in adding the word "test". A user who searches "bitcoin" will see the "signet" entry in their search results and wonder what it is, perhaps even worry about it. Without the word "test" the only two ways to figure out are to google "what is signet" or to launch the application.

I don't expect most users to know what "mainnet" is. They obviously know what Bitcoin is, but they may not know that one or more test networks exist.

**Comment by 1440000bytes** [comment]
> I don't expect most users to know what "mainnet" is. They obviously know what Bitcoin is, but they may not know that one or more test networks exist.

If someone doesnt know what is mainnet they wont use bitcoin core.

**Comment by 1440000bytes** [comment]
Most of the contributors in this repo have no experience about windows including windows servers. Sorry for being rude or straight but its a fact. I love linux but if you support multiple OS which is expected then please understand its different from a user's perspective.

Luke Dashjr will be the last person to ask if I need to add something in core for windows.

I have worked on windows for a long time and love linux. You can ignore my suggestion and NACK. Merge it as other PRs however I know "test" makes no sense there. Signet is enough.

**Comment by TheCharlatan** [comment]
Concept ACK

**Comment by michaelfolkson** [comment]
Concept ACK

Perhaps the Bitcoin Design [community](https://bitcoin.design/) will have a view on the wording. I'd go with standalone "signet" but I haven't done any user research on whether this isn't sufficiently informative.

**Review by maflcko — APPROVED** [review]
Seems fine. Left comments

File: src/Makefile.qt.include [hunk 2]
```diff
@@ -176,6 +176,7 @@ QT_RES_ICONS = \
   qt/res/icons/address-book.png \
   qt/res/icons/bitcoin.ico \
   qt/res/icons/bitcoin_testnet.ico \
+  qt/res/icons/bitcoin_signet.ico \
```
[the remark this answers is not in the document]
Comment (Sjors) [reply src/Makefile.qt.include:179]: Or we could just drop testnet3 :-P

I find the difference in color useful enough.

File: share/setup.nsi.in [hunk 3]
```diff
@@ -141,6 +142,7 @@ Section -un.post UNSEC0001
     Delete /REBOOTOK "$SMPROGRAMS\$StartMenuGroup\Uninstall $(^Name).lnk"
     Delete /REBOOTOK "$SMPROGRAMS\$StartMenuGroup\$(^Name).lnk"
     Delete /REBOOTOK "$SMPROGRAMS\$StartMenuGroup\@PACKAGE_NAME@ (testnet, 64-bit).lnk"
+    Delete /REBOOTOK "$SMPROGRAMS\$StartMenuGroup\@PACKAGE_NAME@ (test signet, 64-bit).lnk"
```
[the remark this answers is not in the document]
Comment (Sjors) [reply share/setup.nsi.in:145]: Done for testnet and signet. I didn't touch `Name "@PACKAGE_NAME@ (64-bit)"` though, not sure if that's safe.

File: src/Makefile.qt.include (hunk 2 above)
[the remark this answers is not in the document]
Comment (Sjors) [reply src/Makefile.qt.include:179]: The incantation is in the commit message. Adding a script makes sense, but I'm not familiar enough with the (native) build tools to know where to add it. This particular command would also add a dependency to imagemagick (if we don't use it already), though I'm sure the operation can be done with other tools if needed.

**Comment by Sjors** [comment]
Creating a shortcut is easy, but creating an icon with a non-confusing color is not.

**Comment by Sjors** [comment]
I would also expect not-super-tech-savy people to be testing applications on custom signets, e.g. testing a vault GUI on Inquisition. I would hope they don't use Windows of course :-)

**Comment by achow101** [comment]
The PR didn't seem to attract much attention in the past. Also, the issue seems not important enough right now to keep it sitting around idle in the list of open PRs.

Closing due to lack of interest.

**Comment by Sjors** [comment]
cc @laanwj

**Comment by laanwj** [comment]
Concept ACK, i think it's important that signet gets its own launch icon if testnet has, it's more useful for testing than testnet. The other option would be to keep only the mainnet icon. But i think there's value in making windows users aware of these kind of things (so they don't reinvent wheels, or, test on mainnet 😅 ).

**Review by BrandonOdiwuor — COMMENTED** [review]
Concept ACK

**Comment by Sjors** [comment]
Rebased for CMake (dropping `src/Makefile.qt.include`) and added a testnet4 shortcut.

**Comment by achow101** [comment]
ACK cfd03de965a081facbd72316c76603dd7aa511bd

Did not test, but the code matches the pattern used for the other networks.

**Review by laanwj — COMMENTED** [review]
\

**Review by laanwj — APPROVED** [review]
ACK cfd03de965a081facbd72316c76603dd7aa511bd

**Review by sedited — APPROVED** [review]
ACK cfd03de965a081facbd72316c76603dd7aa511bd

**Comment by hasti0073** [comment]
1JjXPKN1yyPMxi2WvJv4BHVaJqqgzxSRGg

**Review by hebasto — COMMENTED** [review]
Post-merge ACK cfd03de965a081facbd72316c76603dd7aa511bd.

**Pull request merged**